File: code/python-for-csharp-java/compare_fail/l01_order.cs
// compare_fail/l01_order.cs
Console.WriteLine($"capo {LineTotal("capo", 9.99, 2)}");
Console.WriteLine($"strings {LineTotal("strings", "12.50", 1)}");

static double WithTax(double price) => price + price * 0.15;

static double LineTotal(string product, double price, int quantity)
{
    if (quantity > 10)
    {
        return WithTax(price) * quantity * discount;
    }
    return WithTax(price) * quantity;
}

File: code/python-for-csharp-java/compare_fail/L01Order.java
// compare_fail/L01Order.java
class L01Order {
    static double withTax(double price) {
        return price + price * 0.15;
    }

    static double lineTotal(String product, double price, int quantity) {
        if (quantity > 10) {
            return withTax(price) * quantity * discount;
        }
        return withTax(price) * quantity;
    }

    public static void main(String[] args) {
        System.out.println("capo " + lineTotal("capo", 9.99, 2));
        System.out.println("strings " + lineTotal("strings", "12.50", 1));
    }
}
